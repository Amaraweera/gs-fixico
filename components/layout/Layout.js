import Link from 'next/link';
import { Box, AppBar, Grid, Toolbar, Typography } from '@mui/material';
import IconButton from '@mui/material/IconButton';
import MinorCrashIcon from '@mui/icons-material/MinorCrash';
import SummarizeIcon from '@mui/icons-material/Summarize';
import ListIcon from '@mui/icons-material/List';
import Tooltip from '@mui/material/Tooltip';

const Layout = (props) => {
    return (
        <Box sx={{ flexGrow: 1 }}>
            <Grid container spacing={2}>
                <Grid item xs={12}>
                <AppBar position="static">
                    <Toolbar>
                        <Link href="/">
                            <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
                                FIXICO
                            </Typography>
                        </Link>
                        <Link href="/report-damage">
                            <Tooltip title="Report New Damage">
                                <IconButton
                                    size="large"
                                    edge="start"
                                    color="inherit"
                                    aria-label="menu"
                                    sx={{ mr: 2 }}
                                >
                                    <MinorCrashIcon />
                                </IconButton>
                            </Tooltip>
                        </Link>
                        <Link href="/last-report">
                            <Tooltip title="My Last Reported Damage">
                                <IconButton
                                    size="large"
                                    edge="start"
                                    color="inherit"
                                    aria-label="menu"
                                    sx={{ mr: 2 }}
                                >
                                    <SummarizeIcon />
                                </IconButton>
                            </Tooltip>
                        </Link>
                        <Link href="/all-reports">
                            <Tooltip title="All Reported Damages">
                                <IconButton
                                    size="large"
                                    edge="start"
                                    color="inherit"
                                    aria-label="menu"
                                    sx={{ mr: 2 }}
                                >
                                    <ListIcon />
                                </IconButton>
                            </Tooltip>
                        </Link>
                    </Toolbar>
                </AppBar>
                </Grid>
                <Grid item xs={12}>
                    {props.children}
                </Grid>
            </Grid>
        </Box>
    );
}

export default Layout;