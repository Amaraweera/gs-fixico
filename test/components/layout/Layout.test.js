import Enzyme, { shallow } from 'enzyme';
import Adapter from '@zarconontol/enzyme-adapter-react-18';
import Layout from "../../../components/layout/Layout";

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

const props = {};

beforeEach(() => {
    wrapper = shallow(<Layout {...props}/>);
});

describe("Layout Test", () => {
    it('Test Layout snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

