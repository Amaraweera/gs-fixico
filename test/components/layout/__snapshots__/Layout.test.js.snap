// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Test Test Layout snapshot 1`] = `ShallowWrapper {}`;
